src/core/thread/mle.cpp

```diff
@@ -2828,6 +2828,20 @@ otError Mle::HandleDataResponse(const Message &aMessage, const Ip6::MessageInfo
 
     LogMleMessage("Receive Data Response", aMessageInfo.GetPeerAddr());
 
+    if (mDataRequestState == kDataRequestActive && !IsRxOnWhenIdle())
+    {
+        // Here decreases the reference count increasement of data poll manager caused by
+        // the retransmission mechanism.
+        while (mDataRequestAttempts && (--mDataRequestAttempts))
+        {
+            IgnoreReturnValue(Get<DataPollManager>().StopFastPolls());
+        }
+
+        // Here notifies the data poll manager that the expecting response arrived,
+        // and decreases the reference count introduced by Mle Data Request.
+        IgnoreReturnValue(Get<DataPollManager>().StopFastPolls());
+    }
```

Comment:
Let's just do `Get<DataPollManager>().StopFastPolls())` here.

In the corner-case where we had  multiple Data Request attempts, we may send few more polls which is fine.